lib/StackAnalysis/StackAnalysis.cpp

```diff
@@ -441,6 +1975,244 @@ bool StackAnalysis::runOnModule(Module &M) {
+  }
+  }
+
+  // Default-constructed cache summary for indirect calls
+  FunctionSummary DefaultSummary(model::FunctionType::Values::Regular,
+                                 { ABIRegisters.begin(), ABIRegisters.end() },
+                                 ABIAnalyses::ABIAnalysesResults(),
+                                 {},
+                                 FSO,
+                                 nullptr);
+  FunctionAnalysisResults Properties(std::move(DefaultSummary));
+
+  // Open streams for dumping results
+  std::unique_ptr<raw_fd_ostream> OutputIBI;
+  if (IndirectBranchInfoSummaryPath.getNumOccurrences() == 1) {
+    std::remove(IndirectBranchInfoSummaryPath.c_str());
+    std::error_code EC;
+    OutputIBI = std::make_unique<raw_fd_ostream>(IndirectBranchInfoSummaryPath,
+                                                 EC,
+                                                 llvm::sys::fs::OF_Append);
+    revng_assert(!EC);
+
+    *OutputIBI << "name,ra,fso,address";
+    for (const auto &Reg : ABIRegisters)
+      *OutputIBI << "," << Reg->getName();
+    *OutputIBI << "\n";
+  }
+
+  std::unique_ptr<raw_fd_ostream> OutputAAWriter;
+  if (AAWriterPath.getNumOccurrences() == 1) {
+    std::remove(AAWriterPath.c_str());
+    std::error_code EC;
+    OutputAAWriter = std::make_unique<raw_fd_ostream>(AAWriterPath,
+                                                      EC,
+                                                      llvm::sys::fs::OF_Append);
+    revng_assert(!EC);
+  }
+
+  using CFEPA = CFEPAnalyzer<FunctionAnalysisResults>;
+  CFEPA Analyzer(M,
+                 &GCBI,
+                 Properties,
+                 ABIRegisters,
+                 PreHookMarker,
+                 PostHookMarker,
+                 RetHookMarker,
+                 UnexpectedPCMarker,
+                 *OutputIBI,
+                 *OutputAAWriter);
+
+  // Interprocedural analysis
+  while (!CFEPQueue.empty()) {
+    auto *EntryNode = CFEPQueue.pop();
+    revng_log(StackAnalysisLog,
+              "Analyzing Entry: " << EntryNode->BB->getName());
+
+    // Intraprocedural analysis
+    FunctionSummary Res = Analyzer.analyze(EntryNode->BB);
+    bool Changed = Properties.registerFunction(getBasicBlockPC(EntryNode->BB),
+                                               std::move(Res));
+
+    // Need to recompute the queue?
+    if (Changed) {
+      UniquedQueue<BasicBlockNode *> FakeFunctionWorklist;
+      FakeFunctionWorklist.insert(EntryNode);
+
+      while (!FakeFunctionWorklist.empty()) {
+        auto *Node = FakeFunctionWorklist.pop();
+        for (auto *Caller : Node->predecessors()) {
+          if (Caller == RootNode)
+            break;
+
+          if (!Properties.isFakeFunction(getBasicBlockPC(Caller->BB)))
+            CFEPQueue.insert(Caller);
+          else
+            FakeFunctionWorklist.insert(Caller);
+        }
+      }
+    }
+  }
+
+  // Still OK?
+  revng_assert(llvm::verifyModule(M, &llvm::dbgs()) == false);
```

Comment:
`verifyModule` is kinda expensive. We usually put this under `VerifyLog.isEnabled()`.